File: dp/nth_fibonacci.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution {
public:
  int fib(int N) {
    // code here
    if (N == 0)
      return 0;
    if (N == 1)
      return 1;
    int a = 0, b = 1;
    for (int i = 2; i <= N; ++i) {
      int t = a + b;
      a = b;
      b = t % 10;
    }
    return b;
  }
};

//{ Driver Code Starts.
int main() {
  int t;
  cin >> t;
  while (t--) {
    int N;
    cin >> N;
    Solution ob;
    cout << ob.fib(N) << endl;
  }
  return 0;
}
// } Driver Code Ends

File: dp/nth_fibonacci.py
class Solution:
    def nthFibonacci(self, n: int) -> int:
        # code here
        m = int(1e9 + 7)
        a, b = 1, 1
        for i in range(2, n):
            a, b = a + b, a
            if a > m:
                a -= m
        return a


# {
# Driver Code Starts
if __name__ == "__main__":
    t = int(input())
    for _ in range(t):
        n = int(input())

        obj = Solution()
        res = obj.nthFibonacci(n)

        print(res)


# } Driver Code Ends
